### ISSUE TITLE
Late attachment of Downgraded plugin does not clean up the caches

### ISSUE THREAD

**Issue by meerkat-citronella** [avkonst/hookstate#101]
I am trying to do something very simple and what I consider intuitive... I am trying to create a global state which is an array of objects, and then tap into it later, getting and setting that global state. Problem is I'm not able to get() the value of this state, let alone set() it. 

I have this: 

```
import { createState, useState } from '@hookstate/core'

interface KeyValues {
  [key: string]: string; //e.g. "Date": "7/5/2015"
}

interface KeyValuesByDoc {
  docName: string;
  docType: string;
  docID: string;
  keyValuePairs: KeyValues;
  interpretedKeys: KeyValues;
}

const getKeyValuePairsByDoc = (): KeyValuesByDoc[] => {
...
...
return docData 
}

export const globalDocData = createState(getKeyValuePairsByDoc())

...
...
...

// in a component, 
const docData = useState(globalDocData)

console.log(docData.get())
// => TypeError: docData.get is not a function
```

Now I understand that this isn't necessarily the right way to access the state of an object... even following the docs, the example given doesn't work: 

```
const exampleObjectData = { a: 1, b: 2 }
const globalExampleObject = createState(exampleObjectData)

....
...
...

// in a component 
const exampleState = useState(globalExampleObject)
console.log(exampleState.keys)
// => Proxy{} 
// expected output: [a, b] per the docs
```

Really simple question... how can I 'get' the state value of an array of objects? Can I not just get it... do I have to map it first? Why isn't it as simple as just doing .get() like for primitives? 

I also tried the fix mentioned in [this thread](https://github.com/avkonst/hookstate/issues/92)... 

```
import { Downgraded } from '@hookstate/core'

// in a component
const exampleState = useState(globalExampleObject)
console.log(exampleState.attach(Downgraded).get())
// => TypeError: exampleState.attach is not a function
```

**Comment by avkonst** [comment]
What you do should work.
Are you sure you use useState from Hookstate but not from React package?

**Comment by meerkat-citronella** [comment]
> Are you sure you use useState from Hookstate but not from React package?

Yes. 

The solution I've come up (kinda hacky) is to convert my object to a string when setting, and parsing as JSON after getting... so the whole thing is just a string and behaves with the rules of primitives.

**Comment by avkonst** [comment]
`const exampleState = useState(globalExampleObject)`

What is the result of `Object.keys(globalExampleObject)` and `Object.keys(exampleState)`?

Can you post the screenshot of object inspection for both variables?

Can you create a separate repository with a reproducer?

What you see is abnormal. Something strange with your environment.

**Comment by meerkat-citronella** [comment]
Here is a [code sandbox](https://codesandbox.io/s/youthful-mirzakhani-vni4b?file=/src/App.js) with the issue recreated. I wasn't able to recreate what I'm seeing in my app in the sandbox. 

I realized I was running v2.0.1... upgraded to v3.0.1. This didn't solve the issues, but it prevent them from causing breaking errors. Here are screenshots from my app: 

Creating the state: 
<img width="579" alt="Screen Shot 2020-09-16 at 3 27 44 PM" src="https://user-images.githubusercontent.com/57840235/93383405-9380d000-f831-11ea-90d2-21aff350f811.png">

Using the state in a component: 
<img width="587" alt="Screen Shot 2020-09-16 at 3 27 57 PM" src="https://user-images.githubusercontent.com/57840235/93383441-a4c9dc80-f831-11ea-9a62-6580fbb51b6c.png">
<img width="525" alt="Screen Shot 2020-09-16 at 3 28 13 PM" src="https://user-images.githubusercontent.com/57840235/93383456-aabfbd80-f831-11ea-964b-7515ce487a01.png">
<img width="715" alt="Screen Shot 2020-09-16 at 3 28 45 PM" src="https://user-images.githubusercontent.com/57840235/93383476-b1e6cb80-f831-11ea-9766-0aacbf19ffca.png">

As you can see, the attach() and get() methods are no longer throwing errors, but they are still returning the proxy instead of the actual object.

**Comment by avkonst** [comment]
The sample in codesandbox works fine. And I tested get methods and attached downgraded also makes an effect there. The only weird thing about the codesandbox is that it can not show proxies in the log console properly. It shows null instead. But I guess it is because the browser does not let javascript to inspect objects behind the proxy (which is expected and correct).

If you see different behaviour in you app, could you please commit it to the repository which I could clone and install?

**Comment by meerkat-citronella** [comment]
The code is ensconced in our app and would be difficult to extricate and push to a repo... so, apologies, i can't provide the code for you to review. But thank you for your prompt replies and help with the issue.

**Comment by avkonst** [comment]
I understand why you still see the proxy after the Downgraded(). 

Remove these lines:
![image](https://user-images.githubusercontent.com/3971413/93394767-d3df5d00-f8c8-11ea-987d-253c21fde4a7.png)

And you will see no proxy after downgraded.

The get() caches return value. This might be considered as a bug. I will think if it can be fixed. Current workaround is to attach Downgraded straight after useState call

**Comment by meerkat-citronella** [comment]
This worked! thanks!

**Comment by avkonst** [comment]
Keep it open. I will try to fix it so the order does not change the behaviour

**Comment by TroyJoachim** [comment]
@avkonst 
Were you about to find a possible fix of this. I would like to keep the proxy type on there so I get the performance benefits, then remove it right before I post to my api.

**Comment by avkonst** [comment]
The solution I think is known. Just did not have time to come back to it. I
will have a look at this tomorrow

**Comment by avkonst** [comment]
Fix released in 3.0.6

**Comment by adrianoos** [comment]
Ok, so I used .get({ noproxy: true }) and it works fine now i have array of objects.
